src/app/customer/checkoutSuggestion/BoltCheckoutSuggestion.tsx

```diff
@@ -0,0 +1,94 @@
+import { CheckoutSelectors, CustomerInitializeOptions, CustomerRequestOptions, ExecutePaymentProviderCheckoutOptions } from '@bigcommerce/checkout-sdk';
+import { noop } from 'lodash';
+import React, { memo, useEffect, useState, FunctionComponent, SyntheticEvent } from 'react';
+
+import { TranslatedString } from '../../locale';
+import { Button, ButtonVariant } from '../../ui/button';
+import { IconBolt } from '../../ui/icon';
+
+export interface BoltCheckoutSuggestionProps {
+    isExecutingPaymentProviderCheckout: boolean;
+    methodId: string;
+    deinitializeCustomer(options: CustomerRequestOptions): Promise<CheckoutSelectors>;
+    executePaymentProviderCheckout(options: ExecutePaymentProviderCheckoutOptions): Promise<CheckoutSelectors>;
+    initializeCustomer(options: CustomerInitializeOptions): Promise<CheckoutSelectors>;
+    onUnhandledError?(error: Error): void;
+}
+
+const BoltCheckoutSuggestion: FunctionComponent<BoltCheckoutSuggestionProps> = ({
+    isExecutingPaymentProviderCheckout,
+    methodId,
+    deinitializeCustomer,
+    executePaymentProviderCheckout,
+    initializeCustomer,
+    onUnhandledError = noop,
+}) => {
+    const [ showSuggestion, setShowSuggestion ] = useState<boolean>(false);
+
+    useEffect(() => {
+        deinitializeCustomer({ methodId });
+
+        try {
+            initializeCustomer({
+                methodId,
+                [methodId]: {
+                    onInit: (hasBoltAccount: boolean) => {
+                        setShowSuggestion(hasBoltAccount);
+                    },
+                },
+            });
+        } catch (error) {
+            onUnhandledError(error);
+        }
+
+        return () => {
+            deinitializeCustomer({ methodId });
+        };
+    }, [initializeCustomer, deinitializeCustomer, methodId, onUnhandledError]);
+
+    if (!showSuggestion) {
+        return null;
+    }
+
+    const handleContainerClick = (event: SyntheticEvent)  => {
+        event.stopPropagation();
+    };
+
+    const handleActionClick = async () => {
+        await executePaymentProviderCheckout({ methodId });
+    };
+
+    return (
```

Comment:
It looks like we can create a generic component for suggestion, what do you think?

like we have in payment method https://github.com/bigcommerce/checkout-js/blob/dd430ab979c2583ae01e213e7437238025b4b7a3/src/app/payment/paymentMethod/PaypalCommercePaymentMethod.tsx#L53

Reply:
Thanks! It is a good recommendation 👍
But I think that we need to hold the initiative to that time when we will have more than one provider with the same flow like this to make it more flexible to use